cuqipy_fenics/testproblem.py

```diff
@@ -147,3 +155,275 @@ def form(m,u,p):
+            raise ValueError(
+                "All boundary conditions cannot be Neumann. At least one boundary condition should be Dirichlet.")
+        if bc_values is None:
+            bc_values = [0, 0, 0, 0]
+        elif len(bc_values) != 4:
+            raise ValueError(
+                "The length of bc_values list should be 4. The list should be ordered as follows: [left, bottom, right, top]")
+        bc_values = [to_dolfin_expression(bc_value) for bc_value in bc_values]
+
+        subdomains = self._create_boundaries_subdomains()
+        dirichlet_bcs = self._set_up_dirichlet_bcs(
+            V, bc_types, bc_values, subdomains)
+        neumann_bcs = self._set_up_neumann_bcs(
+            V, bc_types, bc_values, subdomains)
+        adjoint_dirichlet_bcs = self._set_up_adjoint_dirichlet_bcs(
+            V, bc_types, subdomains)
+
+        # Set up the source term
+        if f is None:
+            f = dl.Constant(1)
+        else:
+            f = to_dolfin_expression(f)
+
+        # Set up the variational problem form
+        if mapping is None:
+            def parameter_form(m): return m
+        elif callable(mapping):
+            parameter_form = mapping
+        elif mapping.lower() == 'exponential':
+            def parameter_form(m): return ufl.exp(m)
+        else:
+            raise ValueError('mapping should be a callable, None or a string.')
+
+        def form(m, u, p):
+            return parameter_form(m)*ufl.inner(ufl.grad(u), ufl.grad(p))*ufl.dx\
+                - f*p*ufl.dx\
+                - neumann_bcs(m, p)
+
+        # Create the CUQI PDE object
+        PDE = SteadyStateLinearFEniCSPDE(
+            form,
+            mesh,
+            parameter_function_space=V,
+            solution_function_space=V,
+            dirichlet_bcs=dirichlet_bcs,
+            adjoint_dirichlet_bcs=adjoint_dirichlet_bcs)
+
+        # Create the domain geometry
+        G_FEM = FEniCSContinuous(V)
+        if field_params is None:
+            field_params = {}
+        if field_type is None:
+            G_domain = G_FEM
+        elif field_type == 'KL':
+            if field_params == {}:
+                field_params = {'length_scale': 0.1, 'num_terms': 32}
+            G_domain = MaternExpansion(G_FEM, **field_params)
+        else:
+            raise ValueError('Unknown field type.')
+
+        # Create the range geometry
+        G_range = FEniCSContinuous(V)
+
+        # Create the forward model
+        A = PDEModel(PDE, domain_geometry=G_domain, range_geometry=G_range)
+
+        # Create the prior
+        if prior is None:
+            prior = Gaussian(np.zeros(A.domain_dim), 1,
+                             geometry=G_domain, name='x')
+        elif prior.name != 'x':
+            raise ValueError('Prior name is expected to be "x".')
+
+        # Set up the exact solution
+        if exactSolution is None and field_type == "KL":
+            np.random.seed(15)
```

Comment:
I think this will set the seed globally. We want only to use this fixed seed for the random generation of the exact solution I assume. Please use the new rng style from NumPy to achieve that